File: iOS-Cook-UI/XPUITableViewController.m
//
//  XPUITableViewController.m
//  iOS-Cook-UI
//
//  Created by XP on 3/18/14.
//  Copyright (c) 2014 newland. All rights reserved.
//

#import "XPUITableViewController.h"

static NSString *CellIdentifier = @"Cell";

@interface XPUITableViewController ()

@property (nonatomic, strong) NSArray *uiArray;

@end

@implementation XPUITableViewController

- (id)initWithStyle:(UITableViewStyle)style
{
    self = [super initWithStyle:style];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    // Uncomment the following line to preserve selection between presentations.
    // self.clearsSelectionOnViewWillAppear = NO;
    
    // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
    // self.navigationItem.rightBarButtonItem = self.editButtonItem;
    
    NSURL *path = [[NSBundle mainBundle] URLForResource:@"UIList" withExtension:@"plist"];
    self.uiArray = [NSArray arrayWithContentsOfURL:path];
    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:CellIdentifier];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Table view data source

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    // Return the number of sections.
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    // Return the number of rows in the section.
    return self.uiArray.count;
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:CellIdentifier forIndexPath:indexPath];
    
    // Configure the cell...
    cell.textLabel.text = [self.uiArray objectAtIndex:indexPath.row];
    
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    if ([[self.uiArray objectAtIndex:indexPath.row] isEqualToString:@"Blur Slide Bar"]) {
        [self performSegueWithIdentifier:@"pushBlurSlideBarViewController" sender:self];
    }else if ([[self.uiArray objectAtIndex:indexPath.row] isEqualToString:@"Drawer View Controller"]){
        [self performSegueWithIdentifier:@"pushDrawerViewController" sender:self];
    }else if([[self.uiArray objectAtIndex:indexPath.row] isEqualToString:@"Animate Tab View Controller"]){
        [self performSegueWithIdentifier:@"pushAnimateTabViewController" sender:self];
    }else if ([[self.uiArray objectAtIndex:indexPath.row] isEqualToString:@"Photo Scan"]){
        [self performSegueWithIdentifier:@"pushPhotoBrowseViewController" sender:self];
    }else if([[self.uiArray objectAtIndex:indexPath.row] isEqualToString:@"Zone Refresh"]){
        [self performSegueWithIdentifier:@"pushZoneRefresh" sender:self];
    }else if([[self.uiArray objectAtIndex:indexPath.row] isEqualToString:@"Walk Through"]){
        [self performSegueWithIdentifier:@"pushWalkThroughViewController" sender:self];
    }else if([[self.uiArray objectAtIndex:indexPath.row] isEqualToString:@"Image Processing"]){
        [self performSegueWithIdentifier:@"pushImageProcessingViewController" sender:self];
    }
}

@end

File: iOS-Cook-UI/XPUITableViewController.swift
//
//  XPUITableViewController.swift
//  iOS-Cook-UI
//
//  Created by XP on 10/29/14.
//  Copyright (c) 2014 newland. All rights reserved.
//

import UIKit

class XPUITableViewController: UITableViewController {
    
    var uiArray:[String]!;
    let reuseIdentifier = "reuseIdentifier";

    required init(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder);
    }


    override func viewDidLoad() {
        super.viewDidLoad()

        // Uncomment the following line to preserve selection between presentations
        // self.clearsSelectionOnViewWillAppear = false

        // Uncomment the following line to display an Edit button in the navigation bar for this view controller.
        // self.navigationItem.rightBarButtonItem = self.editButtonItem()
        self.tableView.registerClass(UITableViewCell.self, forCellReuseIdentifier: reuseIdentifier);

        var array: NSArray?
        if let path = NSBundle.mainBundle().pathForResource("UIList", ofType: "plist") {
            array = NSArray(contentsOfFile: path)
            self.uiArray = array as? [String];
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    // MARK: - Table view data source

    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        // #warning Potentially incomplete method implementation.
        // Return the number of sections.
        return 1
    }

    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        // #warning Incomplete method implementation.
        // Return the number of rows in the section.
        if let array = self.uiArray{
            return array.count;
        }else{
            return 0;
        }
    }

    // MARK: table delegate
    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        var tableCell:UITableViewCell;
        if let cell = tableView.dequeueReusableCellWithIdentifier(reuseIdentifier, forIndexPath: indexPath) as? UITableViewCell{
            tableCell = cell;
        }else{
            tableCell = UITableViewCell(style: UITableViewCellStyle.Default, reuseIdentifier: reuseIdentifier);
        }
        tableCell.textLabel.text = self.uiArray?[indexPath.row];

        // Configure the cell...

        return tableCell;
    }
    
    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        tableView.deselectRowAtIndexPath(indexPath, animated: true);
        var cell:UITableViewCell = tableView.cellForRowAtIndexPath(indexPath)!;
        if(cell.textLabel.text == "Table Refresh"){
            var refreshTable:XPRefreshTableViewController = XPRefreshTableViewController();
            self.navigationController?.pushViewController(refreshTable, animated: true);
        }
    }

}
